sim/warlock/affliction/haunt.go

```diff
@@ -0,0 +1,71 @@
+package affliction
+
+import (
+	"time"
+
+	"github.com/wowsims/cata/sim/core"
+	"github.com/wowsims/cata/sim/core/proto"
+	"github.com/wowsims/cata/sim/warlock"
+)
+
+func (affliction *AfflictionWarlock) registerHauntSpell() {
+	if !affliction.Talents.Haunt {
+		return
+	}
+
+	actionID := core.ActionID{SpellID: 48181}
+	debuffMult := core.TernaryFloat64(affliction.HasPrimeGlyph(proto.WarlockPrimeGlyph_GlyphOfHaunt), 1.23, 1.2)
+
+	affliction.HauntDebuffAuras = affliction.NewEnemyAuraArray(func(target *core.Unit) *core.Aura {
+		return target.GetOrRegisterAura(core.Aura{
+			Label:    "Haunt-" + affliction.Label,
+			ActionID: actionID,
+			Duration: time.Second * 12,
+			OnGain: func(aura *core.Aura, sim *core.Simulation) {
+				affliction.AttackTables[aura.Unit.UnitIndex].HauntSEDamageTakenMultiplier *= debuffMult
+			},
+			OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+				affliction.AttackTables[aura.Unit.UnitIndex].HauntSEDamageTakenMultiplier /= debuffMult
+			},
+		})
+	})
+
+	affliction.Haunt = affliction.RegisterSpell(core.SpellConfig{
+		ActionID:       actionID,
+		SpellSchool:    core.SpellSchoolShadow,
+		ProcMask:       core.ProcMaskSpellDamage,
+		Flags:          core.SpellFlagAPL,
+		ClassSpellMask: warlock.WarlockSpellHaunt,
+		MissileSpeed:   20,
+
+		ManaCost: core.ManaCostOptions{
+			BaseCost:   0.12,
+			Multiplier: 1,
+		},
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				GCD:      core.GCDDefault,
+				CastTime: time.Millisecond * 1500,
+			},
+			CD: core.Cooldown{
+				Timer:    affliction.NewTimer(),
+				Duration: time.Second * 8,
+			},
+		},
+
+		DamageMultiplierAdditive: 1,
+		CritMultiplier:           affliction.DefaultSpellCritMultiplier(),
+		ThreatMultiplier:         1,
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			baseDamage := 1078 + (spell.SpellPower()*0.5577)*1.25
```

Comment:
Only the spell power portion of the spell is increased by 25%? This is a bit weird and i'm guessing this is why you're not using the BonusCoefficient field and manually adding it here.

Reply:
This is how the formula is in the tooltip:
![image](https://github.com/wowsims/cata/assets/5125907/5a2d95df-3919-445d-9855-fb70680fef37)

Reply:
Ok so basically you can just put the `BonusCoefficient` to `0.05577*1.25` and not have to do this manual calc in the apply effects